File: laramie-auditoria-webapi/src/domain/entities/incidencia.ts
export default class Incidencia {

    id: number;
	token: string;
	idTipoIncidencia: number;
	idNivelCriticidad: number;
	idUsuario: number;
	fecha: Date;
	idModulo: number;
	origen: string;
	mensaje: string;
	error: Object;
	data: Object;

	constructor(
        id: number = 0,
		token: string = "",
		idTipoIncidencia: number = 0,
		idNivelCriticidad: number = 0,
		idUsuario: number = 0,
		fecha: Date = null,
		idModulo: number = 0,
		origen: string = "",
		mensaje: string = "",
		error: Object = {},
		data: Object = {}
	)
	{
        this.id = id;
		this.token = token;
		this.idTipoIncidencia = idTipoIncidencia;
		this.idNivelCriticidad = idNivelCriticidad;
		this.idUsuario = idUsuario;
		this.fecha = fecha;
		this.idModulo = idModulo;
		this.origen = origen;
		this.mensaje = mensaje;
		this.error = error;
		this.data = data;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.token = row.token ?? "";
		this.idTipoIncidencia = row.idTipoIncidencia ?? 0;
		this.idNivelCriticidad = row.idNivelCriticidad ?? 0;
		this.idUsuario = row.idUsuario ?? 0;
		this.fecha = row.fecha ?? null;
		this.idModulo = row.idModulo ?? 0;
		this.origen = row.origen ?? "";
		this.mensaje = row.mensaje ?? "";
		this.error = row.error ?? {};
		this.data = row.data ?? {};
	}

}

File: modules/auditoria/models/incidencia.py
from sqlalchemy import Column, BigInteger, String, Text, DateTime, func

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Incidencia(Base):
    __tablename__ = "auditoria_incidencias"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    token = Column(String(250), nullable=True)
    id_tipo_incidencia = Column(BigInteger, nullable=False)
    id_nivel_criticidad = Column(BigInteger, nullable=False)
    id_usuario = Column(BigInteger, nullable=True)
    fecha = Column(DateTime, server_default=func.now(), nullable=False)
    id_modulo = Column(BigInteger, nullable=True)
    origen = Column(String(250), nullable=True)
    mensaje = Column(Text, nullable=True)
    error_data = Column(Text, nullable=True)
    data = Column(Text, nullable=True)
